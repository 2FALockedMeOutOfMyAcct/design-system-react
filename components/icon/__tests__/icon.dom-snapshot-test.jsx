/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import Action from '../__examples__/action';
// eslint-disable-next-line camelcase
import UNSAFE_DirectionSettings from '../../utilities/UNSAFE_direction';

const makeRtl = (component) => (
	// eslint-disable-next-line
	<UNSAFE_DirectionSettings.Provider value="rtl">
		<div dir="rtl">{component}</div>
	</UNSAFE_DirectionSettings.Provider>
);

test('Default Action Icon DOM Snapshot', () => {
	const domTree = renderer.create(<Action />).toJSON();
	expect(domTree).toMatchSnapshot();
});

test('Default Action Icon Right to Left (RTL) DOM Snapshot', () => {
	const domTree = renderer.create(makeRtl(<Action />)).toJSON();
	expect(domTree).toMatchSnapshot();
});
