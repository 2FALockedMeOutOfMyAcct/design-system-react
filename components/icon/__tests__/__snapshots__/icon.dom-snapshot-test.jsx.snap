// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Action Icon DOM Snapshot 1`] = `
<span
  className="slds-icon_container slds-icon_container_circle slds-icon-action-description"
>
  <svg
    aria-hidden="true"
    className="slds-icon slds-icon_small"
  >
    <use
      xlinkHref="/assets/icons/action-sprite/svg/symbols.svg#description"
    />
  </svg>
  <span
    className="slds-assistive-text"
  >
    Description of icon
  </span>
</span>
`;

exports[`Default Action Icon Right to Left (RTL) DOM Snapshot 1`] = `
<div
  dir="rtl"
>
  <span
    className="slds-icon_container slds-icon_container_circle slds-icon-action-description"
  >
    <svg
      aria-hidden="true"
      className="slds-icon slds-icon_small"
    >
      <use
        xlinkHref="/assets/icons/action-sprite/svg/symbols-rtl.svg#description"
      />
    </svg>
    <span
      className="slds-assistive-text"
    >
      Description of icon
    </span>
  </span>
</div>
`;
